### ISSUE TITLE
To create a public link, set `share=True` in `launch()`.

### ISSUE THREAD

**Issue by YuWeigang** [CircleRadon/Osprey#11]
Some weights of the model checkpoint at checkpoints/osprey_7b were not used when initializing OspreyLlamaForCausalLM: ['model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.15.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.0.blocks.2.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.12.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.10.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.23.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.18.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.16.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.4.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.7.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.0.blocks.0.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.1.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.1.blocks.2.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.22.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.1.blocks.1.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.17.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.3.blocks.2.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.9.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.0.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.6.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.2.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.20.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.1.blocks.0.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.19.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.13.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.25.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.14.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.24.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.11.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.26.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.8.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.3.blocks.0.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.3.blocks.1.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.5.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.3.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.21.weight', 'model.vision_tower.vision_tower.visual.trunk.stages.0.blocks.1.weight']
- This IS expected if you are initializing OspreyLlamaForCausalLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing OspreyLlamaForCausalLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of OspreyLlamaForCausalLM were not initialized from the model checkpoint at checkpoints/osprey_7b and are newly initialized: ['model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.20.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.2.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.3.blocks.0.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.0.blocks.0.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.14.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.19.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.3.blocks.2.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.16.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.12.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.13.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.7.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.3.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.0.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.15.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.1.blocks.1.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.8.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.10.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.1.blocks.2.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.1.blocks.0.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.18.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.5.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.3.blocks.1.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.26.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.22.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.1.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.9.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.23.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.0.blocks.2.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.0.blocks.1.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.17.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.6.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.4.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.21.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.24.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.25.gamma', 'model.vision_tower.vision_tower.visual.trunk.stages.2.blocks.11.gamma']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Running on local URL:  http://127.0.0.1:8002

To create a public link, set `share=True` in `launch()`.

Does this count as a successful operation? But the access to the result failed at http://127.0.0.1:8002/

**Comment by CircleRadon** [comment]
@YuWeigang  It's right.

**Comment by YuWeigang** [comment]
Hello, thank you very much for your patient responses; I have greatly benefited from them. After running [app.py](http://app.py/) and following the prompts to add share=True, I encountered an error indicating that the Gradio downloaded from the source code cannot be accessed publicly. Do you have a solution for this issue?

**Comment by CircleRadon** [comment]
Yes, we also have encountered this issue, because we add the drawing bounding box function from the official gradio, so `share=True` may not support. However, there is no need to use this, you can try this demo by creating Local Links just like you ran successfully above.
